### PR TITLE
Make wayback url request to use per-profile SharedLoaderFactory

### PR DESCRIPTION
**Pull request by simonhong**
With this factory, traffic to the Internet Archive will use tor
proxy in private(tor). and set cookie disallowed.

Resolves  https://github.com/brave/brave-browser/issues/8419

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Launch tor window
2. Load www.brave.com/bo
3. Capture network request by using wireshark or any other network intercepter
4. Verify that traffic to the Internet Archive(brave-api.archive.org) uses tor proxy
5. With Charles intercepter, it can't capture `brave-api.archive.org` url request.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by kjozwiak** [comment]
Reproduced the original issue on `macOS 10.15.3 x64` using the following build:

Brave | 1.7.22 Chromium: 80.0.3987.122 (Official Build) nightly (64-bit)
--- | ---
Revision | cf72c4c4f7db75bc3da689cd76513962d31c7b52-refs/branch-heads/3987@{#943}
OS | macOS Version 10.15.3 (Build 19D76)

When visiting https://brave.com/bo in a Tor window and selecting `Check for saved version`, you'll notice that https://brave-api.archive.org is being listed in Charles.

Screen Shot 2020-03-02 at 1 53 52 PM

Verification **PASSED** on `macOS 10.15.3 x64` using the following build:

Brave | 1.7.32 Chromium: 80.0.3987.122 (Official Build) nightly (64-bit)
-- | --
Revision | cf72c4c4f7db75bc3da689cd76513962d31c7b52-refs/branch-heads/3987@{#943}
OS | macOS Version 10.15.3 (Build 19D76)

Basically ensured that https://brave-api.archive.org/ isn't being listed in Charles Proxy when using the Tor window. Double checked with @simonhong and ensured that's the expected behaviour.
